File: skills/file_renamer/list_models.py
import google.generativeai as genai

API_KEY = "AIzaSyCW7aJCBWLKQkGfBnC8nhC2jC1UWFXPilY"
genai.configure(api_key=API_KEY)

print("Listing models...")
try:
    for m in genai.list_models():
        if 'generateContent' in m.supported_generation_methods:
            print(f"Name: {m.name}")
except Exception as e:
    print(f"Error: {e}")

File: test_openai_env/list_models.js

const apiKey = "AIzaSyCW7aJCBWLKQkGfBnC8nhC2jC1UWFXPilY"; // Masked

async function main() {
    console.log("Listing available models from Google API...");
    // Standard Gemini API endpoint for listing models
    const url = `https://generativelanguage.googleapis.com/v1beta/models?key=${apiKey}`;

    try {
        const response = await fetch(url);
        if (!response.ok) {
            console.log(`Fetch Status: ${response.status} ${response.statusText}`);
            const text = await response.text();
            console.log(`Body: ${text}`);
        } else {
            const data = await response.json();
            console.log("Models found:");
            if (data.models) {
                data.models.forEach(m => {
                    console.log(`- ${m.name} (DisplayName: ${m.displayName})`);
                    // Check if it supports generateContent
                    if (m.supportedGenerationMethods) {
                        console.log(`  Methods: ${m.supportedGenerationMethods.join(", ")}`);
                    }
                });
            } else {
                console.log("No models returned.");
            }
        }
    } catch (e) {
        console.log("Fetch Error: " + e.message);
    }
}

main();
